src/main/scala/com/fulcrumgenomics/bam/ClipBam.scala

```diff
@@ -76,10 +86,12 @@ class ClipBam
   Io.assertReadable(ref)
   Io.assertCanWriteFile(output)
 
-  validate(clipOverlappingReads || Seq(readOneFivePrime, readOneThreePrime, readTwoFivePrime, readTwoThreePrime).exists(_ != 0),
+  this.clippingMode = this.clippingMode.orElse { if (this.softClip) Some(ClippingMode.Soft) else Some(ClippingMode.Hard) }
```

Comment:
I'd suggest making `this.clippingMode` a `val` and then doing this here instead:
`val mode = this.clippingMode.getOrElse { if (this.softClip) ClippingMode.Soft else ClippingMode.Hard }`